broker/lib/fabrik/Fabrik.js

```diff
@@ -47,12 +47,16 @@ class Fabrik {
         const instance = manager.createInstance(instance_id);
         return Promise
           .try(() => context ? context : instance.platformContext)
-          .then(context => instance.assignPlatformManager(Fabrik.getPlatformManager(context.platform)))
+          .then(context => instance.assignPlatformManager(Fabrik.getPlatformManager(context)))
           .return(instance);
       });
   }
 
-  static getPlatformManager(platform) {
+  static getPlatformManager(context) {
+    let platform = context.platform;
+    if (platform === CONST.PLATFORM.SM) {
+      platform = context.origin;
```

Comment:
any reason why this is being done? Is it because for now for BOTH SM and CF we want to reuse the same platform manager?  
If so then wouldn't it be better to do the below in constants.js and get rid of the above? This way you dont have to revisit this block when you add platformmanager for SM.
```
PLATFORM_MANAGER: {
     'sapcp': 'CfPlatformManager',	     
     'cloudfoundry': 'CfPlatformManager'
...
```

Reply:
In case of SM: origin can be either `cf` or `k8s` hence accordingly it will pick platform manager either `CfPlatformManager` or `K8SPlatformManager`